File: client/src/types/member.ts
export type Member = {
  id: string;
  dateOfBirth: string; // ISO date string (YYYY-MM-DD)
  imageUrl: string;
  userName: string;
  createdAt: string; // ISO datetime string
  lastActive: string; // ISO datetime string
  gender: string;
  description?: string;
  city: string;
  country: string;
};

export type Photo = {
  id: number;
  url: string;
  publicId?: string | null;
  memberId: string;
};

export type EditableMember = {
  userName: string;
  description?: string;
  city: string;
  country: string;
};

export class MemberParams {
  gender?: string;
  minAge = 18;
  maxAge = 100;
  pageNumber = 1;
  pageSize = 10;
  orderBy="lastActive"
}

File: API/Entities/Member.cs


using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{
  public string Id {get;set;} = null!; //! operator suppresses nullable reference type warnings

  public DateOnly DateOfBirth {get;set;}

  public string? ImageUrl {get;set;}

  public required string UserName {get;set;}

  public DateTime CreatedAt {get;set;} = DateTime.UtcNow;

  public DateTime LastActive {get;set;} = DateTime.UtcNow;

  public required string Gender {get;set;}

  public string? Description {get;set;}

  public required string City {get;set;}

  public required string Country {get;set;}



  //Navigation property
  [JsonIgnore] //Do not send photoes when we return member
public List<Photo> Photos {get;set;} =[];


//Like navigation property

[JsonIgnore]
public List<MemberLike> LikedByMembers {get;set;} =[]; //List of users the current users liked by

[JsonIgnore]
public List<MemberLike> LikedMembers {get;set;} = []; //List of users that the current user liked 


  [ForeignKey(nameof(Id))] //Use the Id property in Member as the foreign key for the User navigation property.
  [JsonIgnore]
  public AppUser User {get;set;} = null!;  
}

//Responsibility that this model and table will serve

// Public dating profile

// Search, filters, cards, recommendations

// Frequently read data